services/src/contexts/mod.rs

```diff
@@ -62,19 +62,18 @@ pub trait Context: 'static + Send + Sync + Clone {
 }
 
 pub struct QueryContextImpl {
-    chunk_byte_size: usize,
-}
-
-impl QueryContextImpl {
-    pub fn new(chunk_byte_size: usize) -> Self {
-        Self { chunk_byte_size }
-    }
+    pub chunk_byte_size: usize,
+    pub thread_pool: ThreadPoolContext,
 }
 
 impl QueryContext for QueryContextImpl {
     fn chunk_byte_size(&self) -> usize {
         self.chunk_byte_size
     }
+
+    fn thread_pool(&self) -> ThreadPoolContext {
+        self.thread_pool.clone()
+    }
```

Comment:
Maybe it is better to return a reference to a context. One could clone it if it's necessary but we can skip cloning if we don't need to.

Reply:
b4087ec0a91cee8d5f4773e37be84c6839cd51cc

Reply:
285f84e